docs/usage/design.rst

```diff
@@ -0,0 +1,26 @@
+Design of SCAutolib
+=============================
+
+This section would introduce design decisions that were made during
+implementation with explanation why is it decided in that particular way.
+
+* **SCAutolib uses MVC pattern adopted for CLI and test usage.**
+
+  In earlier versions of the SCAutolib (those, that are not even released) there
```

Comment:
Not sure we need to mention what was wrong with previous versions. I would suggest something like:

"MCV pattern is introduced to improve organization and structure of SCAutolib. Further design decisions respects this pattern giving developers some baseline for structuring of code.